text/0000-assoc-default-groups.md

```diff
@@ -0,0 +1,1078 @@
+
+When an associated type default exists in a `trait` definition,
+it need not be specified in the implementations of that `trait`.
+If implementations of that `trait` do not make that associated type
+available for specialization, the `$default_type` may be assumed
+in other items specified in the implementation.
+If an implementation does make the associated type available for
+further specialization, then other definitions in the implementation
+may not assume the given underlying specified type of the associated type
+and may only assume that it is `Self::TheAsociatedType`.
+
+This applies generally to any item inside a `trait`.
+You may only assume the signature of an item, but not any provided definition,
+in provided definitions of other items.
+For example, this means that you may not assume the value of an
+associated `const` item in other items with provided definition
+in a `trait` definition.
+
+#### Interaction with `existential type`
+
+[RFC 2071]: https://github.com/rust-lang/rfcs/blob/master/text/2071-impl-trait-existential-types.md#reference-existential-types
+
+[RFC 2071] defines a construct `existential type Foo: Bar;` which is permitted
+in associated types and results in an opaque type. This means that the nominal
+type identity is hidden from certain contexts and only `Bar` is extensionally
+known about the type wherefore only the operations of `Bar` is afforded.
+This construct is sometimes written as `type Foo = impl Bar;` in conversation
+instead.
+
+[RFC 1210]: https://github.com/rust-lang/rfcs/blob/master/text/1210-impl-specialization.md#default-impls
+
+With respect to this RFC, the semantics of `type Assoc = impl Bar;`
+inside a trait definition, where `Assoc` is the name of the associated type,
+is understood as what it means in terms of `default impl ..` as discussed
+in [RFC 1210]. What this entails means in concrete terms is that given:
+
+```rust
+trait Foo {
+    type Assoc = impl Bar;
+
+    ...
+}
+```
+
+the underlying type of `Assoc` stays the same for all implementations which
+do not change the default of `Assoc`. The same applies to specializations.
+With respect to type visibility, it is the same as that of `existential type`.
+
+### Specialization groups
+
+Implementations of a `trait` as well as `trait`s themselves may now
+contain *"specialization default groups"* (henceforth: *"groups"*) as
+defined by the [grammar].
+
+Such a group is considered an *atomic unit of specialization*
+and each item in such a group may be specialized / overridden.
+This means that if *one* item is overridden in a group,
+*all* items must be overridden in that group.
+
+Items inside a group may assume the definitions inside the group.
+Items outside of that group may not assume the definitions inside of it.
+
+#### Nesting
+
+There applies no restriction on the nesting of groups.
+This means that you may nest them arbitrarily.
+When nesting does occur, the atomicity applies as if the nesting was flattened.
```

Comment:
s/was flattened/were flattened/.

Wouldn't it be better just to disallow nesting? I thought of this before I read this section, and came to the conclusion that such variations just add needless complexity and potentially confusion.